### PR TITLE
Added integration test

### PR DESCRIPTION
**Pull request by wsowens**
I added an integration test here (`test/test_pipeline.py`) as well as a test data set here ("pOK_barcode"). We can use this both for demonstrations and for automated testing of our image.

### CODE REVIEW

File: test/test_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    output_file = "pipeline/pOK_barcode_test/aggregated-stats.tsv"
+    os.chdir(tmp_path)
+    args = ["make", "-f", str(make_path), str(output_file)]
+    # pylint: disable=subprocess-run-check
+    cmd = subprocess.run(args,
+                         stdout=subprocess.PIPE,
+                         stderr=subprocess.STDOUT)
+    assert cmd.returncode == 0
+    compare_aggregated_stats(
```
Comment (bryanliujiang) [inline test/test_pipeline.py:84]: Sort of pedantic, but I believe the arguments here are flipped.

File: test/test_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    assert all_equal_or_NA(test_df, "BC_Contam_expected", "BC_Contam_actual")
+
+    # Exact LT_10 / LT_3 values fluctuate due to mapping differences
+    assert np.allclose(test_df["LT_10_actual"],
+                       test_df["LT_10_expected"],
+                       rtol=0.01,
+                       equal_nan=True)
+    assert np.allclose(test_df["LT_3_actual"],
+                       test_df["LT_3_expected"],
+                       rtol=0.01,
```
Comment (bryanliujiang) [inline test/test_pipeline.py:48]: Good idea. How did you decide on the `0.01` threshold? Was this empirical?

**Comment by bryanliujiang** [comment]
This is great!! This will definitely simplify QCs.

File: test/test_pipeline.py (hunk 1 above)
Comment (wsowens) [reply test/test_pipeline.py:84]: Not pedantic at all. Everything should be reflexive here but I should stick with the conventions implied by the documentation. I'll push a commit to fix it

File: test/test_pipeline.py (hunk 2 above)
Comment (wsowens) [reply test/test_pipeline.py:48]: Frankly, I don't have a good feel for what the threshold should be. This is just what "feels" right at the moment. If we get a more concrete idea of how LT_3 / LT_10 values fluctuate, we should update this accordingly.

**Review by bryanliujiang — APPROVED** [review]

**Pull request merged**